autogen/agentchat/conversable_agent.py

```diff
@@ -2054,15 +2054,18 @@ async def a_generate_init_message(self, **context) -> Union[str, Dict]:
         self._process_carryover(context)
         return context["message"]
 
-    def register_function(self, function_map: Dict[str, Callable]):
+    def register_function(self, function_map: Dict[str, Union[Callable, None]]):
```

Comment:
If this use case is important, how about add a public method that mirrors `update_tool_signature` and `update_function_signature`:

```python
def update_function_map(self, function_map: Dict[str, Callable], is_remove: bool = False):
  ...
```

And then re-implement the `register_function` to use the `update_function_map`. 

Perhaps we can also put a deprecation notice to `register_function` to gradually route developers to `update_function_map`.

Reply:
The reason for this is that the semantic of `register_function` is of adding new or updating existing functions, not so much of removing functions. So I am wondering if a different public method that more explicitly update the function map is better. 

Just some idea.